### PR TITLE
Remove cugraph-equivariance

### PR DESCRIPTION
**Pull request by BradReesWork**
The cugraph-equivariant package was moved into cuEquivariance a few releases ago.  This PR removes the code and build processes.

### CODE REVIEW

**Review by jameslamb — APPROVED** [review]
Oh wow! Did not realize this was getting cut this release too!

Have you informed ops that we won't be publishing 24.12 packages of this?

There are some other changes that'll be required too:

* devcontainers: remove this: https://github.com/rapidsai/devcontainers/blob/f3b4eba2b485c74d343c4a34fb813d2212129ddb/features/src/rapids-build-utils/opt/rapids-build-utils/manifest.yaml#L233-L235
* rapids-metadata: add a line like https://github.com/rapidsai/rapids-metadata/blob/e8bbce7d5d1b38fe8950a13e3859742d68db729b/src/rapids_metadata/__init__.py#L171

**Review by tingyu66 — APPROVED** [review]

**Review by wangxiaoyunNV — APPROVED** [review]

**Review by alexbarghi-nv — APPROVED** [review]
👍

**Comment by jameslamb** [comment]
I put up devcontainers and rapids-metadata changes for you.

* https://github.com/rapidsai/rapids-metadata/pull/33
* https://github.com/rapidsai/devcontainers/pull/417#issuecomment-2479085423

That devcontainers PR will need to pass and be merged before CI can succeed here.

And before that can happen, we'll need to get https://github.com/rapidsai/cugraph-gnn/pull/69 merged and new `pylibwholegraph` packages built. CI takes 2+ hours there and 3+ hours in the `devcontainers` repo, so best-case it'll be at least 5 hours until CI can pass on this PR.

**Comment by BradReesWork** [comment]
merge conflict caused the nx PR to get merged in

**Review by ChuckHastings — APPROVED** [review]

File: ci/build_wheel.sh [hunk 1]
```diff
@@ -30,26 +30,22 @@ python -m pip wheel \
 sccache --show-adv-stats
 
 # pure-python packages should be marked as pure, and not have auditwheel run on them.
```
Comment (bdice) [inline ci/build_wheel.sh:32]: There are no more pure Python packages in this repo. This comment is obsolete.
```suggestion
```

**Review by bdice — APPROVED** [review]
I fixed an issue in the Python build script and removed some leftover files. The upstream CI should be fixed as well now, so hopefully this round of CI will pass. Thanks, team!

File: cpp/tests/community/balanced_edge_test.cpp [hunk 2]
```diff
@@ -15,7 +15,7 @@
 #include <rmm/device_vector.hpp>
 #include <rmm/exec_policy.hpp>
 
-TEST(balanced_edge, success)
+TEST(balanced_edge, DISABLED_success)
```
Comment (bdice) [inline cpp/tests/community/balanced_edge_test.cpp:18]: @ChuckHastings is currently diagnosing the failure here, which is unrelated to the removal of cugraph-equivariance. We think it is related to https://github.com/rapidsai/raft/pull/2481. I'll open a tracking issue for now.

Comment (bdice) [reply cpp/tests/community/balanced_edge_test.cpp:18]: Tracking issue: https://github.com/rapidsai/cugraph/issues/4777

**Pull request merged**